corehq/apps/hqcase/tests/test_case_helper.py

```diff
@@ -26,11 +27,19 @@ class CaseHelperTests(TestCase):
     def test_create_case_format(self):
         with get_child_case() as case:
             case_data = serialize_case(case)
-            original_case_id = case_data['case_id']
             helper = CaseHelper(domain=DOMAIN)
-            helper.create_case(case_data)
-            self.assertIsInstance(helper.case, CommCareCase)
-            self.assertNotEqual(helper.case.case_id, original_case_id)
+            helper.create_case(case_data, is_serialized=True)
+
+            self.assertNotEqual(helper.case.case_id, case.case_id)
+            self.assertEqual(helper.case.name, case.name)
+            self.assertEqual(helper.case.type, case.type)
+
+    def test_validation_error(self):
+        with get_child_case() as case:
+            case_data = serialize_case(case)
+            helper = CaseHelper(domain=DOMAIN)
+            with self.assertRaises(UserError):
+                helper.create_case(case_data)
```

Comment:
Would it be difficult to make a "generator test" (with [`@generate_cases`](https://github.com/dimagi/commcare-hq/blob/14f30881c46933330194434f5d730eef95bd8e4e/corehq/util/test_utils.py#L403)) that verifies that each of the keys in
```py
        invalid_fields = {
            'case_id',
            'domain',
            'closed',
            'date_closed',
            'date_opened',
            'last_modified',
            'modified_by',
            'server_last_modified',
            'indexed_on',
        }
```
raises an error when it is the only illegal key in `case_data`?

I'm happy to attempt to clarify if this request is not clear.

Reply:
That was a nice way of asking whether I was sure that I had all of the invalid fields :grin: 
e67463f2aa1